contracts/TokenListManager.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {ITokenListManager} from "./ITokenListManager.sol";
+
+contract TokenListManager is ITokenListManager {
+
+    // Storage the token address to whitelist or blacklist
+    address[] private whitelistedTokensArray;
```

Comment:
Generally, wallet will display ERC20 tokens and NFTs in separate pages. So, `TokenList` contract should support to list only ERC20 tokens or NFTs.